jaxb2-maven-plugin fails under Maven 4 classloading: package-info classes not visible in plugin classrealm

## Description

`jaxb2-maven-plugin:3.1.0` (and likely 4.x) fails under Maven 4 with a `ClassNotFoundException` for `org.glassfish.jaxb.core.v2.model.core.package-info`. The JAXB runtime's `package-info` classes are not visible in the plugin's classrealm.

## Error

```
[ERROR] Failed to execute goal org.codehaus.mojo:jaxb2-maven-plugin:3.1.0:schemagen (schemagen) on project optaplanner-core-impl:
  Unable to load the mojo 'schemagen' in the plugin 'org.codehaus.mojo:jaxb2-maven-plugin:3.1.0'.
  A required class is missing: org.glassfish.jaxb.core.v2.model.core.package-info
-----------------------------------------------------
realm =    plugin>org.codehaus.mojo:jaxb2-maven-plugin:3.1.0
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:...jaxb2-maven-plugin-3.1.0.jar
urls[1] = file:...jaxb-xjc-3.0.0.jar
urls[2] = file:...jaxb-jxc-3.0.0.jar
...
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
```

## Root cause

Maven 4's classrealm construction for plugins filters or doesn't expose `package-info` classes from transitive dependencies. The `jaxb-core` jar is either not included in the plugin classrealm or its `package-info` classes are not accessible due to the module/package filtering in `maven.api`.

The same build succeeds with Maven 3, where the plugin classrealm construction is more permissive.

## Reproducer

```bash
git clone https://github.com/apache/incubator-kie-optaplanner.git
cd incubator-kie-optaplanner
mvn -B -e clean package -DskipTests
```

The `schemagen` goal on `optaplanner-core-impl` fails.

## Context

Found during [Maven 4 compatibility testing](https://github.com/gnodet/maven4-testing/issues/13307) of Apache projects.

_Claude Code on behalf of Guillaume Nodet_

--- a/impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/mvnup/goals/PluginUpgradeStrategy.java
+++ b/impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/mvnup/goals/PluginUpgradeStrategy.java
@@ -97,7 +97,12 @@ public class PluginUpgradeStrategy extends AbstractUpgradeStrategy {
                     DEFAULT_MAVEN_PLUGIN_GROUP_ID,
                     "maven-resources-plugin",
                     "3.3.1",
-                    "Beta/RC versions compiled against different Maven 4 API signatures"));
+                    "Beta/RC versions compiled against different Maven 4 API signatures"),
+            new PluginUpgrade(
+                    "org.codehaus.mojo",
+                    "jaxb2-maven-plugin",
+                    "3.2.0",
+                    "Versions before 3.2.0 depend on jaxb-parent:3.0.0 which contains invalid XML rejected by Maven 4"));
 
     private static final List<PluginUpgrade> PLUGIN_DEPENDENCY_UPGRADES = List.of(new PluginUpgrade(
             "org.codehaus.mojo",
@@ -270,6 +275,9 @@ public class PluginUpgradeStrategy extends AbstractUpgradeStrategy {
         upgrades.put(
                 DEFAULT_MAVEN_PLUGIN_GROUP_ID + ":maven-resources-plugin",
                 new PluginUpgradeInfo(DEFAULT_MAVEN_PLUGIN_GROUP_ID, "maven-resources-plugin", "3.3.1"));
+        upgrades.put(
+                "org.codehaus.mojo:jaxb2-maven-plugin",
+                new PluginUpgradeInfo("org.codehaus.mojo", "jaxb2-maven-plugin", "3.2.0"));
         return upgrades;
     }
 

--- a/impl/maven-cli/src/test/java/org/apache/maven/cling/invoker/mvnup/goals/PluginUpgradeStrategyTest.java
+++ b/impl/maven-cli/src/test/java/org/apache/maven/cling/invoker/mvnup/goals/PluginUpgradeStrategyTest.java
@@ -431,6 +431,45 @@ class PluginUpgradeStrategyTest {
         }
 
         @Test
+        @DisplayName("should upgrade jaxb2-maven-plugin when below minimum")
+        void shouldUpgradeJaxb2MavenPluginWhenBelowMinimum() throws Exception {
+            String pomXml = """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <project xmlns="http://maven.apache.org/POM/4.0.0">
+                    <modelVersion>4.0.0</modelVersion>
+                    <groupId>test</groupId>
+                    <artifactId>test</artifactId>
+                    <version>1.0.0</version>
+                    <build>
+                        <plugins>
+                            <plugin>
+                                <groupId>org.codehaus.mojo</groupId>
+                                <artifactId>jaxb2-maven-plugin</artifactId>
+                                <version>3.1.0</version>
+                            </plugin>
+                        </plugins>
+                    </build>
+                </project>
+                """;
+
+            Document document = Document.of(pomXml);
+            Map<Path, Document> pomMap = Map.of(Paths.get("pom.xml"), document);
+
+            UpgradeContext context = createMockContext();
+            UpgradeResult result = strategy.doApply(context, pomMap);
+
+            assertTrue(result.success(), "Plugin upgrade should succeed");
+            assertTrue(result.modifiedCount() > 0, "Should have upgraded jaxb2-maven-plugin");
+
+            Editor editor = new Editor(document);
+            Element root = editor.root();
+            String version = root.path("build", "plugins", "plugin", "version")
+                    .map(Element::textContentTrimmed)
+                    .orElse(null);
+            assertEquals("3.2.0", version);
+        }
+
+        @Test
         @DisplayName("should not upgrade when version is already higher")
         void shouldNotUpgradeWhenVersionAlreadyHigher() throws Exception {
             String pomXml = """

